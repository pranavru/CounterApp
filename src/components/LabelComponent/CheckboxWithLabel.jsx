/* eslint-disable react/prop-types */
/* eslint-disable jsx-a11y/label-has-associated-control */
/* eslint-disable react/no-access-state-in-setstate */
/* eslint-disable react/destructuring-assignment */
// CheckboxWithLabel.js

import React from 'react';

export default class CheckboxWithLabel extends React.Component {
  constructor(props) {
    super(props);
    this.state = { isChecked: false };
    this.onChange = this.onChange.bind(this);
  }

  onChange() {
    const checkedState = this.state.isChecked;
    this.setState({ isChecked: !checkedState });
  }

  render() {
    return (
      <label>
        <input
          type="checkbox"
          checked={this.state.isChecked}
          onChange={this.onChange}
        />
        {this.state.isChecked ? this.props.labelOn : this.props.labelOff}
      </label>
    );
  }
}
