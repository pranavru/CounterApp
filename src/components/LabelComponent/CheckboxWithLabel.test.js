/* eslint-disable react/jsx-filename-extension */
/* eslint-disable import/no-unresolved */
/* eslint-disable import/extensions */
// __tests__/CheckboxWithLabel-test.js
import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup, fireEvent, render } from '@testing-library/react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import CheckboxWithLabel from './CheckboxWithLabel';

// Note: running cleanup afterEach is done automatically for you in
// @testing-library/react@9.0.0 or higher
// unmount and cleanup DOM after the test is finished.
configure({ adapter: new Adapter() });
afterEach(cleanup);
describe('Checkbox with Label', () => {
  const component = renderer.create(
    <CheckboxWithLabel labelOn="On" labelOff="Off" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // Renders the Checkbox Label component using react testing library
  it('CheckboxWithLabel changes the text after click', () => {
    const { queryByLabelText, getByLabelText } = render(
      <CheckboxWithLabel labelOn="On" labelOff="Off" />,
    );

    expect(queryByLabelText(/off/i)).toBeTruthy();

    fireEvent.click(getByLabelText(/off/i));

    expect(queryByLabelText(/on/i)).toBeTruthy();
  });

  // Render the Checkbox label component using react enzyme shallow library
  test('should render using enzyme Checkbox should initialize to Off', () => {
    const checkbox = shallow(<CheckboxWithLabel labelOff="Off" labelOn="On" />);
    expect(checkbox.text()).toBe('Off');
  });

  test('Checkbox should set to On onClick', () => {
    const checkbox = shallow(<CheckboxWithLabel labelOff="Off" labelOn="On" />);
    checkbox.find('input').simulate('change');
    expect(checkbox.text()).toBe('On');
  });
});
