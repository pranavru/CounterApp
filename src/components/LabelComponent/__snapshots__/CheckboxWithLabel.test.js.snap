// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<label>
  <input
    checked={false}
    onChange={[Function]}
    type="checkbox"
  />
  Off
</label>
`;
